modules/rhcos-load-firmware-blobs.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="rhcos-load-firmware-blobs_{context}"]
+
+= Loading custom firmware blobs in the machine config manifest
+
+Because the default location for firmware blobs in `/usr/lib` is read-only, you can locate a custom firmware blob by updating the search path. This enables you to load local firmware blobs in the machine config manifest when the blobs are not managed by {op-system}.
+
+.Procedure
+
+. Create a Butane config file, `98-worker-firmware-blob.bu`, that updates the search path so that it is root-owned and writable to local storage. The following example places the custom blob file from your local workstation onto nodes under `/var/lib/firmware`.
++
+[NOTE]
+====
+See "Creating machine configs with Butane" for information about Butane.
+====
+.Butane config file for custom firmware blob
++
+[source,yaml]
+----
+variant: openshift
+version: 4.9.0
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 98-worker-firmware-blob
+storage:
+  files:
+  - path: /var/lib/firmware/<package_name> <1>
+    contents:
+      local: <package_name> <2>
+    mode: 0644 <3>
+openshift:
+  kernel_arguments:
+    - 'firmware_class.path=/var/lib/firmware' <4>
+----
++
+<1> Sets the path on the node where the firmware package is copied to.
+<2> Specifies a file with contents that are read from a local file directory on the system running Butane. The path of the local file is relative to a `files-dir` directory, which must be specified by using the `--files-dir` option with Butane in the following step.
+<3> Sets the permissions for the file on the {op-system} node. It is recommended to set `0644` permissions.
+<4> The `firmware_class.path` parameter customizes the kernel search path of where to look for the custom firmware blob that was copied from your local workstation onto the root file system of the node. This example uses `/var/lib/firmware` as the customized path.
+
+. Run Butane to generate a `MachineConfig` object file that uses a copy of the firmware blob on your local workstation named `98-worker-firmware-blob.yaml` that contains the configuration to be delivered to the nodes. The following example uses the `--files-dir` option to specify the directory on your workstation where the local file or files are located:
```

Comment:
Optional: The first sentence read a bit long-winded in my opinion. Perhaps we could break it into two sentences? Maybe something like this?

```
Run Butane to generate a `MachineConfig` object file that uses a copy of the firmware blob
on your local workstation named `98-worker-firmware-blob.yaml`. The firmware blob contains
the configuration to be delivered to the nodes.
```

Reply:
Agreed and resolved